### PR TITLE
Login

### PR DESCRIPTION
**Pull request by zMingGit**

### CODE REVIEW

File: seahub/api2/endpoints/account.py [hunk 1]
```diff
@@ -39,6 +40,7 @@ def get_account_info(user):
     info['is_staff'] = user.is_staff
     info['is_active'] = user.is_active
     info['create_time'] = user.ctime
+    info['loginname'] = profile.login_id if profile else ''
```
Comment (xiez) [inline seahub/api2/endpoints/account.py]: login_id

File: setenv.sh.template [hunk 2]
```diff
@@ -1,3 +1,5 @@
-export CCNET_CONF_DIR=/home/plt/dev/ccnet/seafile/tests/basic/conf2
-export SEAFILE_CONF_DIR=/home/plt/dev/ccnet/seafile/tests/basic/conf2/seafile-data
-export PYTHONPATH=/opt/lib/python2.6/site-packages:thirdpart:$PYTHONPATH
+export CCNET_CONF_DIR=/home/zming/dev/conf
```
Comment (xiez) [inline setenv.sh.template]: 这个文件不要提交

File: seahub/templates/sysadmin/userinfo.html [hunk 3]
```diff
@@ -351,6 +374,43 @@ <h2 class="alc">{% trans "This user has not created or joined any groups" %}</h2
     return false;
 });
 
+$('#set-loginid-form').submit(function() {
+    var loginid = $.trim($('[name="loginid"]', $(this)).val());
+    var $department = $('#loginid');
```
Comment (xiez) [inline seahub/templates/sysadmin/userinfo.html]: $loginid

File: seahub/api2/endpoints/account.py [hunk 4]
```diff
@@ -108,10 +110,22 @@ def _update_account_additional_info(self, request, email):
             profile = Profile.objects.get_profile_by_user(email)
             if profile is None:
                 profile = Profile(user=email)
-
             profile.nickname = name
             profile.save()
 
+        loginid = request.data.get("login_id", None)
```
Comment (xiez) [inline seahub/api2/endpoints/account.py]: loginid = request.data.get("login_id", '').strip()

File: seahub/api2/endpoints/account.py [hunk 5]
```diff
@@ -108,10 +110,22 @@ def _update_account_additional_info(self, request, email):
...
+        if loginid is not None:
```
Comment (xiez) [inline seahub/api2/endpoints/account.py]: if loginid != '':

File: seahub/api2/endpoints/account.py (hunk 4 above)
Comment (zMingGit) [inline seahub/api2/endpoints/account.py]: get it

File: seahub/api2/endpoints/account.py (hunk 5 above)
Comment (zMingGit) [inline seahub/api2/endpoints/account.py]: get it

File: seahub/api2/endpoints/account.py [hunk 6]
```diff
@@ -165,6 +175,18 @@ def put(self, request, email, format=None):
                 return api_error(status.HTTP_400_BAD_REQUEST,
                         _(u"Name should not include '/'."))
 
+        #argument check for loginid
+        loginid = request.data.get("login_id", None)
+        if loginid is not None:
+            loginid = loginid.strip()
+            if loginid == "":
+                return api_error(status.HTTP_400_BAD_REQUEST,
+                            _(u'can not use space to be loginid'))
```
Comment (xiez) [inline seahub/api2/endpoints/account.py]: "Login id can't be empty"

File: seahub/api2/endpoints/account.py [hunk 7]
```diff
@@ -165,6 +175,18 @@ def put(self, request, email, format=None):
...
+            profilebyloginid = Profile.objects.get_username_by_login_id(loginid)
+            if profilebyloginid is not None:
+                return api_error(status.HTTP_400_BAD_REQUEST,
+                          _(u'loginid is be used'))
```
Comment (xiez) [inline seahub/api2/endpoints/account.py]: "Login id %s already exists." % loginid

File: seahub/templates/sysadmin/userinfo.html [hunk 8]
```diff
@@ -351,6 +374,48 @@ <h2 class="alc">{% trans "This user has not created or joined any groups" %}</h2
...
 });
 
+$('#set-loginid-form').submit(function() {
+    var loginid = $.trim($('[name="loginid"]', $(this)).val());
+    var $loginid = $('#loginid');
+    var $error = $('.error', $(this));
+    var $submitBtn = $('[type="submit"]', $(this));
+    disable($submitBtn);
+    if (loginid == ''){
+        $error.html("can't use space to name").show();
```
Comment (xiez) [inline seahub/templates/sysadmin/userinfo.html]: 同上

File: seahub/templates/sysadmin/userinfo.html [hunk 9]
```diff
@@ -98,6 +110,13 @@
             <input type="submit" value="{% trans "Submit" %}" class="submit" />
         </form>
 
+        <form id="set-loginid-form" method="put" action="" class="hide">{% csrf_token %}
```
Comment (llj) [inline seahub/templates/sysadmin/userinfo.html]: `method="put"`

File: seahub/templates/sysadmin/userinfo.html [hunk 10]
```diff
@@ -351,6 +374,48 @@ <h2 class="alc">{% trans "This user has not created or joined any groups" %}</h2
...
 
+$('#set-loginid-form').submit(function() {
+    var loginid = $.trim($('[name="loginid"]', $(this)).val());
+    var $loginid = $('#loginid');
+    var $error = $('.error', $(this));
+    var $submitBtn = $('[type="submit"]', $(this));
+    disable($submitBtn);
+    if (loginid == ''){
+        $error.html("Login id can't be empty").show();
+        enable($submitBtn);
```
Comment (llj) [inline seahub/templates/sysadmin/userinfo.html]: line 382 - line 385

File: seahub/templates/sysadmin/userinfo.html [hunk 11]
```diff
@@ -351,6 +374,48 @@ <h2 class="alc">{% trans "This user has not created or joined any groups" %}</h2
...
+    $.ajax({
+        url: '{% url 'api2-account' email %}',
+        type: 'PUT',
+        dataType: 'json',
+        cache: false,
+        beforeSend: prepareCSRFToken,
+        data: {'login_id': loginid},
+        success: function(data) {
+            if (data['login_id'] == '') {
+                $loginid.html('--')
```
Comment (llj) [inline seahub/templates/sysadmin/userinfo.html]: 上面两行

**Pull request merged**